src/pybind/rados/setup.py

```diff
@@ -1,51 +1,187 @@
+# Bump this version, after every changeset
+# NOTE: This version is not the same as get_ceph_version()
+
+__version__ = '0.0.1'
+
+
+def get_ceph_version():
     try:
-        for line in open(os.path.join(os.path.dirname(__file__), "..", "ceph_ver.h")):
+        for line in open(os.path.join(os.path.dirname(__file__), "..", "..", "ceph_ver.h")):
             if "CEPH_GIT_NICE_VER" in line:
-                return line.split()[2][1:-1]
+                return line.split()[2].strip('"')
         else:
             return "0"
     except IOError:
         return "0"
 
-class EggInfoCommand(egg_info):
-    def finalize_options(self):
-        egg_info.finalize_options(self)
-        if "build" in self.distribution.command_obj:
-            build_command = self.distribution.command_obj["build"]
-            self.egg_base = build_command.build_base
-            self.egg_info = os.path.join(self.egg_base, os.path.basename(self.egg_info))
+
+def get_python_flags():
+    cflags = {'I': [], 'extras': []}
+    ldflags = {'l': [], 'L': [], 'extras': []}
+
+    if os.environ.get('VIRTUAL_ENV', None):
+        python = "python"
+    else:
+        python = 'python' + str(sys.version_info.major) + '.' + str(sys.version_info.minor)
+
+    python_config = python + '-config'
+
+    for cflag in subprocess.check_output(
+            [python_config, "--cflags"]
+    ).strip().decode('utf-8').split():
+        if cflag.startswith('-I'):
+            cflags['I'].append(cflag.replace('-I', ''))
+        else:
+            cflags['extras'].append(cflag)
+
+    for ldflag in subprocess.check_output(
+            [python_config, "--ldflags"]
+    ).strip().decode('utf-8').split():
+        if ldflag.startswith('-l'):
+            ldflags['l'].append(ldflag.replace('-l', ''))
+        if ldflag.startswith('-L'):
+            ldflags['L'].append(ldflag.replace('-L', ''))
+        else:
+            ldflags['extras'].append(ldflag)
+
+    return {
+        'cflags': cflags,
+        'ldflags': ldflags
+    }
+
+
+def check_sanity():
+    """
+    Test if development headers and library for rados is available by compiling a dummy C program.
+    """
+
+    tmp_dir = tempfile.mkdtemp(dir=os.path.dirname(__file__))
+    tmp_file = os.path.join(tmp_dir, 'rados_dummy.c')
+
+    with open(tmp_file, 'w') as fp:
+        dummy_prog = textwrap.dedent("""
+        #include <rados/librados.h>
+
+        int main(void) {
+            rados_t cluster;
+            rados_create(&cluster, NULL);
+            return 0;
+        }
+        """)
+        fp.write(dummy_prog)
+
+    compiler = new_compiler()
+    customize_compiler(compiler)
+
+    if set(['MAKEFLAGS', 'MFLAGS', 'MAKELEVEL']).issubset(set(os.environ.keys())):
+        # The setup.py has been invoked by a top-level Ceph make.
+        # Set the appropriate CFLAGS and LDFLAGS
+        CEPH_ROOT = subprocess.check_output(['git', 'rev-parse', '--show-toplevel']).strip().decode('utf-8')
```

Comment:
@jdurgin I will make the changes. Thanks.

Regarding the build output dir, it can be an environment variable exported just like `CYTHON_BUILD_DIR`.